src/ControllerStatusIO.c

```diff
@@ -751,3 +762,86 @@ int Ros_Controller_GetActiveAlarmCodes(USHORT active_alarms[MAX_ALARM_COUNT + MA
 
     return num_entries;
 }
+
+/**
+ * Attempt to load on-controller extrinsic kinematic calibration data to improve
+ * accuracy of TF broadcasts (for frames for which such calibration data exists).
+ *
+ * NOTE: returns TRUE IFF at least one calibration file loaded successfully,
+ *       FALSE in all other cases.
+*/
+static BOOL Ros_Controller_LoadGroupCalibrationData(Controller* const controller)
+{
+    BOOL bOneCalibFileLoaded = FALSE;
+
+    for (int i = 0; i < MAX_ROBOT_CALIBRATION_FILES; i += 1)
+    {
+        MP_RB_CALIB_DATA calibData;
+        if (Ros_mpGetRobotCalibrationData(i, &calibData) == OK)
+        {
+            Ros_Debug_BroadcastMsg("%s: file %d loaded OK", __func__, i);
+
+            //Take note at least one calibration file loaded successfully.
+            //
+            //NOTE: successful loading of a single calibration file will of
+            //course not mean all groups (that would benefit from it) are actually
+            //calibrated correctly. But this is the best we can do, as there is
+            //no requirement to calibrate every group to every other group, nor
+            //will this prevent MotoROS2 from successfully controlling motion.
+            //
+            //TF frame broadcasts for uncalibrated groups will however be incorrect,
+            //which is why posting an alarm in case there are no calibration files
+            //at all on a multigroup system is considered a good thing to do.
+            bOneCalibFileLoaded = TRUE;
+
+            if (calibData.s_rb.grp_no <= MP_R8_GID && //the slave is a robot
+                calibData.m_rb.grp_no <= MP_R8_GID) //the master is another robot's RF
+            {
+                int groupIndex = mpCtrlGrpId2GrpNo((MP_GRP_ID_TYPE)calibData.s_rb.grp_no);
+                MP_COORD* coord = &controller->ctrlGroups[groupIndex]->robotCalibrationToBaseFrame;
+                coord->x = calibData.pos_uow[0];
+                coord->y = calibData.pos_uow[1];
+                coord->z = calibData.pos_uow[2];
+                coord->rx = calibData.ang_uow[0];
+                coord->ry = calibData.ang_uow[1];
+                coord->rz = calibData.ang_uow[2];
+            }
+        }
+    }
+
+    return bOneCalibFileLoaded;
+}
+
+//TODO(gavanderhoorn): should be able to make this 'static' once Yaskawa-Global/motoros2#173
```

Comment:
I don't understand this comment. Why can't it be static now?

Reply:
`static` functions are local to the compilation unit. The test calling this function is not in this compilation unit. If it were `static`, we could not call it from elsewhere.